api/src/main/java/com/github/retrooper/packetevents/protocol/world/chunk/palette/DataPalette.java

```diff
@@ -186,7 +186,7 @@ private static Palette createPalette(int bitsPerEntry, PaletteType paletteType)
         }
     }
 
-    private static int index(int x, int y, int z) {
-        return y << 8 | z << 4 | x;
+    private static int index(PaletteType paletteType, int x, int y, int z) {
+        return (y << paletteType.getBitShift() | z) | x << paletteType.getBitShift();
```

Comment:
this is incorrect, y should be shifted two times the bit shift
otherwise y and x would "occupy" the same bits of the index

Reply:
Yes, I made a mistake. I pushed for a commit and would you please check it.